PlaywrightSolar/PageObjects/SolarPanelsPage.cs

```diff
@@ -0,0 +1,128 @@
+    {
+        await Assertions.Expect(PageTitle).ToBeVisibleAsync();
+    }
+    
+    public async Task OpenProductFilter()
+    {
+        await ProductFilterButton.ClickAsync();
+    }
+
+    public async Task FilterByManufacturer(string manufacturer)
+    {
+        var filterLocators = new Dictionary<string, ILocator>
+        {
+            { "Abi-Solar", AbiSolarFilter },
+            { "C&T Solar", CAndTSolarFilter },
+            { "JA Solar", JaSolarFilter },
+            { "Jinko Solar", JinkoSolarFilter },
+            { "SOLA", SolaFilter },
+            { "Ulica Solar", UliсaFilter },
+            { "Yingli Solar", YingliFilter }
+        };
+
+        if (filterLocators.TryGetValue(manufacturer, out var filterLocator))
+        {
+            var elementsBeforeFilter = await page.QuerySelectorAllAsync(".col.s12.m6.l4.xl3.prod-holder");
+            var textsBeforeFilter =
+                await Task.WhenAll(elementsBeforeFilter.Select(async element => await element.InnerTextAsync()));
+
+            await filterLocator.ClickAsync();
+            await Assertions.Expect(filterLocator).ToBeCheckedAsync();
+
+            await page.WaitForTimeoutAsync(2000);
+
+            var elementsAfterFilter = await page.QuerySelectorAllAsync(".col.s12.m6.l4.xl3.prod-holder");
+            var textsAfterFilter =
+                await Task.WhenAll(elementsAfterFilter.Select(async element => await element.InnerTextAsync()));
+
+            Assert.That(textsBeforeFilter, Is.Not.EquivalentTo(textsAfterFilter), "Product filter did not work, elements are the same");
+        }
+    }
+
+    public async Task<bool> ManufacturerFilterResult(string manufacturer)
+    {
+        await Task.Delay(2000);
+        var results = await page.QuerySelectorAllAsync(".col.s12.m6.l4.xl3.prod-holder");
+
+        Assert.That(results.Any(), $"No elements found for manufacturer: {manufacturer}");
+
+        foreach (var result in results)
+        {
+            var title = await result.InnerTextAsync();
+            if (!title.Contains(manufacturer, StringComparison.OrdinalIgnoreCase))
+            {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    public async Task FilterByPanelType(string panelType)
+    {
+        var filterLocators = new Dictionary<string, ILocator>
+        {
+            { "Монокристал", MonoCrystalPanelTypeFilter },
+            { "Полікристал", PolyCrystalPanelTypeFilter }
+        };
+
+        if (filterLocators.TryGetValue(panelType, out var filterLocator))
+        {
+            await filterLocator.ClickAsync();
+        }
+    }
+
+    public async Task<bool> PanelTypeFilterResult(string panelType)
+    {
+        await Task.Delay(2000);
+
+        var parentElements = await page.QuerySelectorAllAsync(".col.s12.m6.l4.xl3.prod-holder");
+
+        Assert.That(parentElements.Any(), $"No elements found for panel type: {panelType}");
+
+        foreach (var parent in parentElements)
+        {
```

Comment:
all this logic should be in the Test. Page may return just list of product names